blocks/library/gallery/block.js

```diff
@@ -56,13 +57,26 @@ class GalleryBlock extends Component {
 	}
 
 	onSelectImage( index ) {
-		return () => {
+		return ( event ) => {
+			// ignore clicks in the editable caption
+			if ( event.target.tagName === 'FIGCAPTION' ) {
+				return;
+			}
 			this.setState( ( state ) => ( {
 				selectedImage: index === state.selectedImage ? null : index,
 			} ) );
 		};
 	}
 
+	onFocusImageCaption( index ) {
+		return ( focusValue ) => {
+			this.setState( {
+				selectedImage: index,
+			} );
+			this.props.setFocus( focusValue );
```

Comment:
I wonder how this works. the `setFocus` should be passed an object. Maybe something like this `{ index }` to keep track of the focused caption.

Reply:
Hi @youknowriad, normally we pass an object in the format { index: val, ... focusValue }, but in this case, we already had a mechanism to track the active image so we pass only focus value and we make use of the existing mechanism focus={ this.state.selectedImage === index && focus ? focus : undefined }. I think adding a new property to the focus value to track the active image would be redundant in this case given that we already have a mechanism for that.

Reply:
Ok, so this means, if you edit the caption of the image, it's selected. (not against this) but I can't test for now as the TinyMCE CDN is broken.

Reply:
Yes, that's right when we try to edit a caption we automatically select that image.